src/psyclone/psyir/transformations/intrinsics/matmul2code_trans.py

```diff
@@ -257,28 +318,14 @@ def validate(self, node, options=None):
         else:
             # There should be one index per dimension. This is enforced
             # by the array create method so is not tested here.
-
-            # The first two indices should be ranges. This is a
-            # limitation of this transformation, not the PSyIR, as it
-            # would be valid to have any two indices being
-            # ranges. Further, this transformation is currently
-            # limited to Ranges which specify the full extent of the
-            # dimension.
-            if not (matrix1.is_full_range(0) and matrix1.is_full_range(1)):
+            # For matrix1, we must have exactly 2 full ranges.
+            full_range_order, _, _ = _get_full_range_split(matrix1)
+            n_full_ranges = len(full_range_order)
+            if n_full_ranges != 2:
                 raise TransformationError(
-                    f"To use matmul2code_trans on matmul, the first two "
-                    f"indices of the 1st argument '{matrix1.name}' must be "
-                    f"full ranges.")
-
-            if len(matrix1.children) > 2:
-                # The 3rd index and onwards must not be ranges.
-                for (count, index) in enumerate(matrix1.children[2:]):
-                    if isinstance(index, Range):
-                        raise TransformationError(
-                            f"To use matmul2code_trans on matmul, only the "
-                            f"first two indices of the 1st argument are "
-                            f"permitted to be Ranges but found "
-                            f"{type(index).__name__} at index {count+2}.")
+                    f"To use matmul2code_trans on matmul, exactly two "
```

Comment:
Not yours but while we're here would you mind replacing "matmul2code_trans" with "{self.name}".